### PR TITLE
Fix Pickup Locking when ev.IsAllowed = false

### PR DESCRIPTION
**Pull request by rayzerbrain**
**PickingUpItemEventArgs.IsAllowed = false** caused the Pickup to permanently lock for some reason. Resetting aspects of the PickupSyncInfo seems to solve the problem. Tested by myself but could be problems I'm unaware of.

~~Not Invalid IL code I promise~~

### CODE REVIEW

**Comment by NaoUnderscore** [comment]
"Avoiding the problem" is not a valid solution.
I'm pretty sure you're gonna break CustomItems.
Also, this could lead to many issues, and since you also said "but could be problems I'm unaware of" you already know that this "solution" is gonna lead to many issues instead.
Before approving it I need to make sure it doesn't break anything, although I'm sure it's not the right way to cope this problem.
I'd like to handle this issue, because I'm sure that I can fix it, but as I already said before, this is not the right way to solve this issue.

**Comment by rayzerbrain** [comment]
That makes sense to me, I'll try poking around a bit more for a better solution, unless you want to take care of it instead?

**Comment by rayzerbrain** [comment]
Don't ask me why this works, but it does.

It seems redundant but I did lots of testing and this is the easiest way it seems to work.

File: Exiled.Events/Patches/Events/Player/PickingUpItem.cs [hunk 1]
```diff
@@ -37,6 +37,7 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
             Label cdcLabel = generator.DefineLabel();
 
             LocalBuilder ev = generator.DeclareLocal(typeof(PickingUpItemEventArgs));
+            LocalBuilder syncInfo = generator.DeclareLocal(typeof(PickupSyncInfo));
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Player/PickingUpItem.cs:40]: Remove it

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

**Comment by SebasCapo** [comment]
When testing these changes, make sure they don't cause #886 to happen again! 😅

**Comment by rayzerbrain** [comment]
That's actually the exact issue I'm trying to fix. Either the NW code changed in the past month to reintroduce the bug or it's only happening in specific situations or something. Can you verify that you currently do or do not have this issue?

**Comment by joker-119** [comment]
The description of your fix and what the code actually does do not match. You're not spawning a new item here anywhere.

**Comment by rayzerbrain** [comment]
> The description of your fix and what the code actually does do not match. You're not spawning a new item here anywhere.

That was my previous method of fixing the problem. My new solution is monkeying around with the PickupBase info, which somehow works

**Comment by joker-119** [comment]
My point was, you should update your first comment to reflect what's actually going on since they no longer match.

**Review by joker-119 — APPROVED** [review]

**Review by NaoUnderscore — APPROVED** [review]

**Pull request merged**